packages/fether-react/src/utils/withHealth.js

```diff
@@ -142,93 +127,39 @@ export default compose(
+          const isGood =
+            isSync && !isNoPeers && isClockSync && isNodeConnected && online;
+          let payload;
+
+          // Status - list of all states of health store
+          let status = {
+            internet: online, // Internet connection
+            nodeConnected: isNodeConnected, // Connected to local Parity Ethereum node
+            clockSync: isClockSync, // Local clock is not synchronised
+            downloading: isDownloading, // Currently downloading Parity Ethereum
+            launching: !isApiConnected, // Launching Parity Ethereum only upon startup
+            peers: !isNoPeers, // Connecion to peer nodes
+            syncing: !isSync, // Synchronising blocks
+            good: isGood // Synchronised and no issues
+          };
 
-          // No connection to the internet and not connected to node
-          if (isNoInternetAndNoNodeConnected) {
-            return {
-              ...props,
-              health: {
-                status: STATUS.NO_NODE_CONNECTED_AND_NO_INTERNET
-              }
-            };
-          }
-
-          // Parity is being downloaded
+          // Payload - optional payload of a state
           if (isDownloading) {
-            return {
-              ...props,
-              health: {
-                status: STATUS.DOWNLOADING,
-                payload: {
-                  percentage: new BigNumber(Math.round(downloadProgress * 100))
-                }
-              }
-            };
-          }
-
-          // Parity is being launched
-          if (!isApiConnected) {
-            return {
-              ...props,
-              health: {
-                status: STATUS.LAUNCHING
-              }
-            };
-          }
-
-          // At this point we have a successful connection to parity
-
-          // No connection to the internet and but connected to node
-          if (isNoInternetAndNodeConnected) {
-            return {
-              ...props,
-              health: {
-                status: STATUS.NODE_CONNECTED_AND_NO_INTERNET
-              }
-            };
-          }
-
-          // Clock is not synchronized
-          if (!isClockSync) {
-            return {
-              ...props,
-              health: {
-                status: STATUS.NO_CLOCK_SYNC
-              }
-            };
-          }
-
-          // Not enough peers
-          if (isNoPeers) {
-            return {
-              ...props,
-              health: {
-                status: STATUS.NO_PEERS
-              }
-            };
-          }
-
-          // Syncing blocks
-          if (!isSync) {
-            return {
-              ...props,
-              health: {
-                status: STATUS.SYNCING,
-                payload: syncPayload
-              }
+            payload = {
+              syncPercentage: new BigNumber(Math.round(downloadProgress * 100))
             };
+          } else if (!isSync) {
+            payload = syncPayload;
```

Comment:
What if we're syncing and downloading at the same time? Which payload will be taken into account?

I think in this case payload should also be an object whose keys are the keys in status, and values can be anything

Reply:
Good idea, I've pushed a commit with the change